packages/pattern-picker/src/item.tsx

```diff
@@ -11,6 +12,70 @@ type Props = {
 	pattern: Pattern;
 };
 
+type ParsedMetaData = {
+	siteTitle?: string;
+	siteTagline?: string;
+	siteLogoUrl?: string;
+};
+
+const parsePatternMetaData = ( patternMeta: PatternMeta ): ParsedMetaData => {
+	const result: ParsedMetaData = {};
+
+	const getPatternMetaValue = ( patternMetaKey: string, regexp: RegExp ) => {
+		const [ , value ] = patternMetaKey.split( regexp );
+		if ( value ) {
+			// We have to unescape `,` because the pattern meta is separated with commas and we store the escaped value
+			return value.replace( /%2C/g, ',' );
+		}
+
+		return null;
+	};
+
+	Object.keys( patternMeta ).forEach( ( patternMetaKey ) => {
+		const siteTitle = getPatternMetaValue( patternMetaKey, /^site_title_/ );
+		const siteTagline = getPatternMetaValue( patternMetaKey, /^site_tagline_/ );
+		const siteLogoUrl = getPatternMetaValue( patternMetaKey, /^site_logo_url_/ );
+
+		if ( siteTitle ) {
+			result.siteTitle = siteTitle;
+		}
+
+		if ( siteTagline ) {
+			result.siteTagline = siteTagline;
+		}
+
+		if ( siteLogoUrl ) {
+			result.siteLogoUrl = siteLogoUrl;
```

Comment:
If you keep this in kebab case, you can pass them directly to the URLSearchParams like shown below.